### DISCUSSION TITLE
osu!taiko Hit Animations Inconsistent in Editor

### DISCUSSION THREAD

**Discussion by zachmanthethird** [ppy/osu#17095] [Q&A]
Version 2022.226.0
When seeking through a beatmap, hit animations don't load as you'd expect. Additionally, the hit target freezes when rewinding.

https://user-images.githubusercontent.com/74262407/156725840-cf22db46-d870-425e-b99e-6a9e27c122df.mp4

Additionally, "flicking" with your mouse wheel will have the same effect:

https://user-images.githubusercontent.com/74262407/156728668-59187347-4f88-4c94-8784-3555679a9a8a.mp4

While it hasn't happened recently, some longer maps (>8 minutes) sometimes only animate hit animations every fourth note or so. You *could* replicate this behavior by abusing right-arrow seeking, but I don't think that qualifies as a normal use case.

**Comment by peppy** [comment]
This is going to be the same issue as https://github.com/ppy/osu/issues/10909, which is caused by large seeks in conjunction with frame stability being turned off in the editor.

A potential way forward I just thought of may be to enforce frame stability for the `DrawableScrollingRuleset.TimeRange` to ensure the visible portion looks correct.

**Reply by bdach** [reply]
Wouldn't enabling frame stability for the visible time range imply enabling frame stability... period? Like if forward smooth seeks are to be frame stable, then you have to play out everything in between and that's just normal frame stability. Similarly even with instant seeks you'd need full frame stability for this not to rear its head again.

**Reply by peppy** [reply]
You might be right.. I'd still need to test whether it would help at all.
